File: frontend/src/lib/api/onboarding.ts
import api, { apiPost } from "@/lib/api";

// ---- Request / Response Types ----

export interface OnboardRequest {
  area_code?: string;
}

export interface OnboardResponse {
  workspace_id: string;
  agent_id: string;
  phone_number_id: string | null;
  phone_number: string | null;
  /** False when no SMS number was auto-provisioned — campaigns can't launch yet. */
  phone_provisioned: boolean;
  google_calendar_connected: boolean;
  message: string;
}

export interface LaunchCampaignResponse {
  campaign_id: string;
  campaign_name: string;
  campaign_status: string;
  contacts_imported: number;
  contacts_skipped: number;
  contacts_failed: number;
  phone_number_used: string;
  agent_id: string;
  started_at: string | null;
}

// ---- API Functions ----

export function onboard(data: OnboardRequest): Promise<OnboardResponse> {
  return apiPost<OnboardResponse>("/api/v1/onboarding/onboard", data);
}

export function createCampaignFromCsv(
  workspaceId: string,
  file: File,
  options: {
    skipDuplicates?: boolean;
    campaignName?: string;
    areaCode?: string;
  } = {},
): Promise<LaunchCampaignResponse> {
  const formData = new FormData();
  formData.append("file", file);
  if (options.skipDuplicates !== undefined) {
    formData.append("skip_duplicates", String(options.skipDuplicates));
  }
  if (options.campaignName) {
    formData.append("campaign_name", options.campaignName);
  }
  if (options.areaCode) {
    formData.append("area_code", options.areaCode);
  }

  return api
    .post<LaunchCampaignResponse>(`/api/v1/onboarding/campaigns`, formData, {
      headers: { "Content-Type": "multipart/form-data" },
    })
    .then((r) => r.data);
}

File: backend/app/schemas/onboarding.py
"""Onboarding schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class OnboardRequest(BaseModel):
    """Request body for the onboarding endpoint."""

    area_code: str | None = Field(
        None,
        min_length=3,
        max_length=3,
        pattern=r"^\d{3}$",
        description="Optional 3-digit US area code for phone number provisioning",
    )


class OnboardResponse(BaseModel):
    """Response from the onboarding endpoint."""

    workspace_id: uuid.UUID
    agent_id: uuid.UUID
    phone_number_id: uuid.UUID | None
    phone_number: str | None
    phone_provisioned: bool = Field(
        ...,
        description=(
            "True when an SMS-capable phone number was provisioned during onboarding. "
            "When false, the workspace cannot launch SMS/voice campaigns until a number is added."
        ),
    )
    google_calendar_connected: bool
    message: str


class LaunchCampaignResponse(BaseModel):
    """Response from the onboarding CSV-to-campaign endpoint."""

    campaign_id: uuid.UUID
    campaign_name: str
    campaign_status: str
    contacts_imported: int
    contacts_skipped: int
    contacts_failed: int
    phone_number_used: str
    agent_id: uuid.UUID
    started_at: datetime | None
